marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowMigrator.java

```diff
@@ -0,0 +1,191 @@
+                        File stepFile = targetDir.resolve(stepFileName).toFile();
+                        logger.info("Creating step artifact "+ stepFile.toString());
+                        if (stepFile.exists()) {
+                            String msg = "Step artifact " + stepFile.toString() + " already exists. The step artifact will be written to ";
+                            //Update step artifact with new name
+                            String stepName = new StringBuilder(flow.getName()).append("-").append(step.getName()).toString();
+                            newStepArtifact.put("name", stepName);
+                            //Update the filename
+                            stepFileName = new StringBuilder(flow.getName()).append("-").append(stepFileName).toString();
+                            stepFile = targetDir.resolve(stepFileName).toFile();
+
+                            logger.info(msg + stepFile.toString()) ;
+                        }
+                        try{
+                            writer.writeValue(stepFile, newStepArtifact);
+                            logger.info("Step artifact " + stepFile.toString() + " successfully created.");
+                        }
+                        catch(IOException e){
+                            logger.error("Step artifact " + stepFile.toString() + " creation failed: " + e.getMessage());
+                        }
+                    }
+                    else {
+                        logger.info("Step " + step.getName() + " is not an out of the box ingestion or mapping step. It will remain inline inside the flow artifact");
+                        newSteps.set(entry.getKey(), mapper.valueToTree(step));
+                    }
+                }
+                newFlow.set("steps",newSteps);
+                File flowFile = Paths.get(hubProject.getFlowsDir().toString(), flow.getName() + FlowManager.FLOW_FILE_EXTENSION).toFile();
+                try{
+                    writer.writeValue(flowFile, newFlow);
+                    logger.info("Flow " + flowFile.toString() + " successfully migrated.");
+                }
+                catch(IOException e){
+                    logger.error("Flow artifact " + flowFile.toString() + " creation failed: " + e.getMessage());
+                }
+            }
+        });
+    }
+
+    // Only create step artifacts for ootb ingestion and mapping steps. Other steps(including custom ingestion and mapping)
+    // will be inline
+    private boolean stepRequiresMigration(Step step) {
+        return (StepDefinition.StepDefinitionType.MAPPING.equals(step.getStepDefinitionType()) && "entity-services-mapping".equalsIgnoreCase(step.getStepDefinitionName())) ||
+            (StepDefinition.StepDefinitionType.INGESTION.equals(step.getStepDefinitionType()) &&  "default-ingestion".equalsIgnoreCase(step.getStepDefinitionName()));
```

Comment:
ingestion step seems to be show up as "Unknown" in the the flow, since its expecting a loadData config

Reply:
This has to be verified after DHFPROD-4951 and DHFPROD-4952 are merged